packages/lending/src/adapter.ts

```diff
@@ -0,0 +1,1051 @@
+      });
+      collateralSwapLogics.push(addLogic);
+    }
+
+    // ---------- withdraw ----------
+    const withdrawLogic = await protocol.newWithdrawLogic({
+      output: new common.TokenAmount(wrappedSrcToken, srcAmount),
+      marketId,
+      account,
+    });
+    collateralSwapLogics.push(withdrawLogic);
+
+    afterPortfolio.withdraw(wrappedSrcToken, srcAmount);
+
+    // ---------- flashloan repay ----------
+    collateralSwapLogics.push(flashLoanRepayLogic);
+
+    // ---------- tx related ----------
+    const estimateResult = await apisdk.estimateRouterData(
+      {
+        chainId: this.chainId,
+        account,
+        logics: collateralSwapLogics,
+      },
+      this.permitType
+    );
+
+    const buildRouterTransactionRequest = (
+      args?: Omit<apisdk.RouterData, 'chainId' | 'account' | 'logics'>,
+      apiKey?: string
+    ): Promise<common.TransactionRequest> =>
+      apisdk.buildRouterTransactionRequest(
+        { ...args, chainId: this.chainId, account, logics: collateralSwapLogics },
+        apiKey ? { 'x-api-key': apiKey } : undefined
+      );
+
+    return {
+      fields: {
+        srcToken: srcToken,
+        srcAmount: srcAmount,
+        destToken: destToken,
+        // TODO: check which one is right
+        destAmount: swapQuotation.output.amount, //withdrawLogic.fields.output.amount,
+        portfolio,
+        afterPortfolio,
+      },
+      estimateResult,
+      buildRouterTransactionRequest,
+      logics: collateralSwapLogics,
+    };
+  }
+
+  // 1. flashloan destToken
+  // 2. swap destToken to srcToken
+  // 3. repay srcToken
+  // 4. borrow destToken
+  // 5. flashloan repay destToken
+  // * srcToken = old borrow token
+  // * destToken = new borrow token
+  async getDebtSwap(
+    protocolId: string,
+    marketId: string,
+    params: BaseParams,
+    account: string,
+    portfolio?: Portfolio
+  ): Promise<BaseFields> {
+    const { srcAmount } = params;
+    const srcToken = common.classifying(params.srcToken);
+    const destToken = common.classifying(params.destToken);
+    const wrappedSrcToken = srcToken.wrapped;
+    const wrappedDestToken = destToken.wrapped;
+
+    const debtSwapLogics: apisdk.Logic<any>[] = [];
+    const protocol = this.getProtocol(protocolId);
+
+    portfolio = portfolio || (await protocol.getPortfolio(account, marketId));
+    const afterPortfolio = portfolio.clone();
+
+    // ---------- Pre-calc quotation ----------
+    const swapper = this.findSwapper([wrappedSrcToken, wrappedDestToken]);
+    // get the quotation for how much dest token is needed to exchange for the src amount
+    let swapQuotation = await swapper.quote({
+      tokenIn: wrappedDestToken,
+      output: { token: wrappedSrcToken, amount: srcAmount },
+      slippage: defaultSlippage,
+    });
+    // convert swap type to exact in
+    swapQuotation = await swapper.quote({
+      input: swapQuotation.input,
+      tokenOut: srcToken,
+      slippage: defaultSlippage,
```

Comment:
still trying to understand how this exact in works:

lets say srcAmount = 1000, srcToken:destToken = 1:1, defaultSlippage = 500 bps (5%)
at Ln.310, I will spend 1050 destTokens if 5% slippage is reached.
at 316, I will spend 1050 destTokens but only get 1050*0.95 = 997.5 srcTokens (if 5% slippage is reached).

Though it still works for the user, the repaid amount becomes 997.5 not 1000.
I'm wondering if we need to pay attention to this. Also, please let me know if any of the calculation is wrong. Thx.